xmtp/README.md

```diff
@@ -108,7 +127,53 @@ init():
     return
 ```
 
-### Receiving an invite
+### Receiving invites
+
+```plaintext
+downloadInvites():
+    Get the `refresh_jobs` record with an id of `invites`, and obtain a lock on the row:
+        Store `now()` in memory to mark the job execution start
+        Fetch all messages from invite topic with timestamp > refresh_job.last_run - PADDING_TIME # PADDING TIME accounts for eventual consistency of network. Maybe 30s.
+        For each message in topic:
+            Save (or ignore if already exists) raw message to inbound_invite table with status of PENDING
+        Update `refresh_jobs` record last_run = current_timestamp
+
+processInvites():
+    For each inbound_invite in PENDING state:
+        If the payload is malformed and the proto cannot be decoded:
+            Set inbound_invite state to INVALID
+            continue
+        If an existing session exists with the `inviter.installation_id`:
+            Decrypt the inner invite using the existing session
+            If decryption fails:
+                Set inbound_invite state to DECRYPTION_FAILURE
+                continue
+            Update session in the database
+        else:
+            Create a new inbound session with the inviter
+            Decrypt the inner invite using the new session
```

Comment:
Hmm. Good point.

If we chose to overwrite it in this race condition, we could run into a case where the two installations actually swap sessions (since both would have non-matching invites and would overwrite their session with the other's). That's just as bad as doing nothing.

But having to support multiple sessions would mean we would need some sort of `session_id` on all messages to disambiguate.

Think I'll look into this some more. Olm must have run into something similar and had to resolve it somehow.